### PR TITLE
CIFY-85: Upgraded deps to fix security issues and added types

### PR DESCRIPTION
**Pull request by lnewson**
This updates the deps to fix the security alerts and I've added strict types throughout. I had done the types a few weeks back when I was messing with a PowerPaste ticket, so I thought I might as well include them.

### CODE REVIEW

File: src/main/ts/ephox/photon/Navigation.ts [hunk 1]
```diff
@@ -1,21 +1,26 @@
-import { document } from '@ephox/dom-globals';
+import { Document, document, Element as DomElement, Node } from '@ephox/dom-globals';
 import { Option } from '@ephox/katamari';
 import { Element, Traverse } from '@ephox/sugar';
 
-var view = function (doc) {
+export interface Navigation<E, D> {
+  view: (doc: D) => Option<E>;
```
Comment (tinydylan) [inline src/main/ts/ephox/photon/Navigation.ts:6]: Fields can be readonly

File: src/main/ts/ephox/photon/Reader.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-var iframeDoc = function (element) {
-  var dom = element.dom();
+const iframeDoc = (element: Element<HTMLIFrameElement>): Option<Element<Document>> => {
+  const dom = element.dom();
   try {
-    var idoc = dom.contentWindow ? dom.contentWindow.document : dom.contentDocument;
-    return idoc !== undefined && idoc !== null ? Option.some(Element.fromDom(idoc)) : Option.none();
+    const idoc = dom.contentWindow ? dom.contentWindow.document : dom.contentDocument;
+    return Option.from(idoc).map(Element.fromDom);
```
Comment (tinydylan) [inline src/main/ts/ephox/photon/Reader.ts:9]: FYI there's an `Options.mapFrom`, if you prefer. Either way is fine, though.

File: src/main/ts/ephox/photon/Reader.ts [hunk 3]
```diff
@@ -15,15 +15,11 @@ var iframeDoc = function (element) {
...
-var doc = function (element) {
-  var optDoc = iframeDoc(element);
-  return optDoc.fold(function () {
-    return element;
-  }, function (v) {
-    return v;
-  });
+const doc = (element: Element<HTMLIFrameElement>) => {
+  const optDoc = iframeDoc(element);
+  return optDoc.fold(() => Traverse.owner(element), (v) => v);
```
Comment (tinydylan) [inline src/main/ts/ephox/photon/Reader.ts:20]: I think you could use `getOrThunk` here.

File: src/main/ts/ephox/photon/Navigation.ts (hunk 1 above)
Comment (lnewson) [reply src/main/ts/ephox/photon/Navigation.ts:6]: I need to remember to include that more, thanks for the catch!

File: src/main/ts/ephox/photon/Reader.ts (hunk 3 above)
Comment (lnewson) [reply src/main/ts/ephox/photon/Reader.ts:20]: Yeah we can, I didn't really look at those sorts of things, but I'll fix them up anyways.

File: src/main/ts/ephox/photon/Reader.ts (hunk 2 above)
Comment (lnewson) [reply src/main/ts/ephox/photon/Reader.ts:9]: Oh, I didn't know that existed, that's handy thanks!

**Pull request merged**